src/psyclone/tests/psyir/nodes/whileloop_test.py

```diff
@@ -0,0 +1,173 @@
+
+''' Performs py.test tests on the WhileLoop PSyIR node. '''
+
+from __future__ import absolute_import
+import pytest
+from psyclone.core import VariablesAccessInfo
+from psyclone.errors import InternalError, GenerationError
+from psyclone.psyir.backend.fortran import FortranWriter
+from psyclone.psyir.nodes import Assignment, BinaryOperation, Literal, \
+                                 Reference, Return, Schedule, WhileLoop
+from psyclone.psyir.nodes.node import colored
+from psyclone.psyir.symbols import DataSymbol, REAL_SINGLE_TYPE, BOOLEAN_TYPE
+from psyclone.tests.utilities import check_links
+
+
+def test_whileloop_invalid_annotation():
+    ''' Test that initialising WhileLoop with invalid annotations produces the
+    expected error.'''
+
+    with pytest.raises(InternalError) as err:
+        _ = WhileLoop(annotations=["invalid"])
+    assert ("WhileLoop with unknown annotation 'invalid', valid "
+            "annotations are:") in str(err.value)
+
+
+def test_whileloop_node_str():
+    ''' Check the node_str method of the WhileLoop class.'''
+
+    coloured_loop = colored("WhileLoop", "red")
+
+    loop = WhileLoop()
+    output = loop.node_str()
+    assert coloured_loop+"[]" in output
+
+    loop = WhileLoop(annotations=['was_unbounded'])
+    output = loop.node_str()
+    assert coloured_loop+"[annotations='was_unbounded']" in output
+
+
+def test_whileloop_create_and_refence_accesses():
+    '''Test that the create method in the WhileLoop class correctly creates
+    a WhileLoop instance and that the reference_access method correctly
+    captures variable accesses.'''
+
+    ref1 = Reference(DataSymbol("tmp", REAL_SINGLE_TYPE))
+    ref2 = Reference(DataSymbol("pmt", REAL_SINGLE_TYPE))
+    loop_condition = BinaryOperation.create(BinaryOperation.Operator.GT, ref1,
+                                            Literal("0.0", REAL_SINGLE_TYPE))
+    loop_body = [Assignment.create(ref2, Literal("1.0", REAL_SINGLE_TYPE))]
+    loop = WhileLoop.create(loop_condition, loop_body)
+    loop_schedule = loop.children[1]
+    assert isinstance(loop_schedule, Schedule)
+    check_links(loop, [loop_condition, loop_schedule])
+    check_links(loop_schedule, loop_body)
+    result = FortranWriter().whileloop_node(loop)
+    assert result == ("do while (tmp > 0.0)\n"
+                      "  pmt = 1.0\n"
+                      "end do\n")
+    var_access_info = VariablesAccessInfo()
+    loop.reference_accesses(var_access_info)
+    assert (str(var_access_info)) == "pmt: WRITE, tmp: READ"
+
+
+def test_whileloop_create_invalid():
+    '''Test that the create method in the WhileLoop class raises the expected
+    exception if the provided input is invalid.'''
+
+    loop_condition = Literal('true', BOOLEAN_TYPE)
+    loop_body = [Assignment.create(
+        Reference(DataSymbol("tmp", REAL_SINGLE_TYPE)),
+        Literal("0.0", REAL_SINGLE_TYPE)),
+               Assignment.create(
```

Comment:
For the purposes of this test I think you only need a single, valid Node in the loop_body - there's no need to go to the trouble of creating a second assignment.

Reply:
Done.